### PR TITLE
Slurm installation and setup tutorial.

### PR DESCRIPTION
**Pull request by zbenta**
Added a tutorial on how to install and setup slurm.

### CODE REVIEW

File: slurm/slurm.md [hunk 1]
```diff
@@ -0,0 +1,941 @@
...
+#COMPUTE NODES
+NodeName=wn001 NodeAddr=X.X.X.X  CPUs=2 Sockets=1 CoresPerSocket=2 ThreadsPerCore=1 State=IDLE RealMemory=2048
+
+
+```
+
+Create a /etc/slurm/cgroup.conf
+
+```
+CgroupAutomount=yes
```
Comment (Obihoernchen) [inline slurm/slurm.md]: `CgroupAutomount` was removed in Slurm [23.11](https://github.com/SchedMD/slurm/blob/slurm-23.11/RELEASE_NOTES#L104-L107). We might want to comment it out by default?

File: slurm/slurm.md [hunk 2]
```diff
@@ -0,0 +1,941 @@
...
+ConstrainCores=yes
+ConstrainRAMSpace=yes
+ConstrainSwapSpace=yes
+TaskAffinity=no
```
Comment (Obihoernchen) [inline slurm/slurm.md]: `TaskAffinity` was removed in Slurm [22.05](https://github.com/SchedMD/slurm/blob/slurm-22.05/RELEASE_NOTES#L92).

File: slurm/slurm.md [hunk 3]
```diff
@@ -0,0 +1,941 @@
...
+
```
Comment (Obihoernchen) [inline slurm/slurm.md]: Maybe add `ConstrainDevices=yes` by default?

File: slurm/slurm.md [hunk 4]
```diff
@@ -0,0 +1,941 @@
...
+Setup user munge and slurm
+
+```
+export MUNGEUSER=981
+groupadd -g $MUNGEUSER munge
+useradd  -m -c "MUNGE Uid 'N' Gid Emporium" -d /var/lib/munge -u $MUNGEUSER -g munge  -s /sbin/nologin munge
+export SlurmUSER=982
+groupadd -g $SlurmUSER slurm
+useradd  -m -c "Slurm workload manager" -d /var/lib/slurm -u $SlurmUSER -g slurm  -s /bin/bash slurm
```
Comment (Obihoernchen) [inline slurm/slurm.md]: I think groups and users should be system groups (add `-r`).
Slurm user should have `/sbin/nologin` shell, too.

File: slurm/slurm.md [hunk 5]
```diff
@@ -0,0 +1,941 @@
...
+
+scp /etc/slurm/cgroup.conf root@shadow.local:/etc/slurm/.
+
+```
+
+Enable and start the slurmctld:
+
+```
+systemctl enable slurmctld
+systemctl start slurmctld
```
Comment (Obihoernchen) [inline slurm/slurm.md]: `systemctl enable --now slurmctld`

File: slurm/slurm.md [hunk 6]
```diff
@@ -0,0 +1,941 @@
...
+ConstrainSwapSpace=yes
+TaskAffinity=no
+
+```
+
+Enable and start the slurmctld:
+
+```
+systemctl enable slurmctld
+systemctl start slurmctld
```
Comment (Obihoernchen) [inline slurm/slurm.md:633]: `systemctl enable --now slurmctld`

File: slurm/slurm.md [hunk 7]
```diff
@@ -0,0 +1,941 @@
...
+
+```
+scp /etc/munge/munge.key root@wn001.local:/etc/munge/.
+```
+
+enable and start munge service
+
+```
+systemctl enable munge
+systemctl start munge
```
Comment (Obihoernchen) [inline slurm/slurm.md:693]: `systemctl enable --now munge`

File: slurm/slurm.md [hunk 8]
```diff
@@ -0,0 +1,941 @@
...
+chown munge:munge /etc/munge/munge.key
+```
+
+
+Enable and and start slurmd service:
+
+```
+systemctl enable slurmd
+
+systemctl start slurmd
```
Comment (Obihoernchen) [inline slurm/slurm.md:722]: `systemctl enable --now slurmd`

File: slurm/slurm.md [hunk 9]
```diff
@@ -0,0 +1,941 @@
...
+
+```
+chown munge:munge /etc/munge/munge.key
+```
+
+enable and start munge service
+
+```
+systemctl enable munge
+systemctl start munge
```
Comment (Obihoernchen) [inline slurm/slurm.md:775]: `systemctl enable --now munge`

File: slurm/slurm.md [hunk 10]
```diff
@@ -0,0 +1,941 @@
...
+
+```
+exportfs -a
+
+```
+
+Enable and start the nfs service:
+```
+systemctl enable nfs
+systemctl start nfs
```
Comment (Obihoernchen) [inline slurm/slurm.md:124]: `systemctl enable --now nfs`

File: slurm/slurm.md [hunk 11]
```diff
@@ -0,0 +1,941 @@
...
+chmod 400 /etc/munge/munge.key
+
+```
+
+#### Enable and start munge service
+
+```
+systemctl enable munge
+
+systemctl start munge
```
Comment (Obihoernchen) [inline slurm/slurm.md:219]: `systemctl enable --now munge`

File: slurm/slurm.md [hunk 12]
```diff
@@ -0,0 +1,941 @@
...
+
+```
+chown munge:munge /etc/munge/munge.key
+```
+
+enable and start munge service
+
+```
+systemctl enable munge
+systemctl start munge
```
Comment (Obihoernchen) [inline slurm/slurm.md:316]: `systemctl enable --now munge`

File: slurm/slurm.md [hunk 13]
```diff
@@ -0,0 +1,941 @@
...
```
Comment (Obihoernchen) [inline slurm/slurm.md]: `systemctl enable --now munge`

File: slurm/slurm.md [hunk 14]
```diff
@@ -0,0 +1,941 @@
...
+```
+mmkdir /var/log/slurm
+chown -R slurm:slurm /var/log/slurm
+```
+
+Start the slurmdbd service
+
+```
+systemctl enable slurmdbd
+systemctl start slurmdbd
```
Comment (Obihoernchen) [inline slurm/slurm.md]: `systemctl enable --now slurmdbd`

File: slurm/slurm.md [hunk 15]
```diff
@@ -0,0 +1,941 @@
...
+
+Setup user munge and slurm
+
+```
+export MUNGEUSER=981
+groupadd -g $MUNGEUSER munge
+useradd  -m -c "MUNGE Uid 'N' Gid Emporium" -d /var/lib/munge -u $MUNGEUSER -g munge  -s /sbin/nologin munge
+export SlurmUSER=982
+groupadd -g $SlurmUSER slurm
+useradd  -m -c "Slurm workload manager" -d /var/lib/slurm -u $SlurmUSER -g slurm  -s /bin/bash slurm
```
Comment (Obihoernchen) [inline slurm/slurm.md]: See above.

File: slurm/slurm.md [hunk 16]
```diff
@@ -0,0 +1,941 @@
...
+
+```
+scp /etc/slurm/cgroup.conf root@ui101.local:/etc/slurm/.
+                                                            
+scp /etc/slurm/slurm.conf root@ui101.local:/etc/slurm/.
+```
+
+Install slurm:
+```
+yum --enablerepo=Slurm_Repo install slurm-pam_slurm slurm-libpmi slurm-slurmd slurm-contribsy
```
Comment (Obihoernchen) [inline slurm/slurm.md]: `slurm-contribsy` typo?

File: slurm/slurm.md [hunk 17]
```diff
@@ -0,0 +1,941 @@
...
+SlurmdLogFile=/var/log/slurmd.log
+JobAcctGatherFrequency=30
+JobAcctGatherType=jobacct_gather/cgroup
+SlurmctldDebug=info
+SlurmdDebug=info
+SlurmSchedLogFile=/var/log/slurmSched.log
+SlurmSchedLogLevel=1
+#
+#Energy consumption test
+AcctGatherEnergyType=acct_gather_energy/rapl
```
Comment (Obihoernchen) [inline slurm/slurm.md]: Maybe add comment that this can be very wrong for some newer Intel CPUs. Check: https://bugs.schedmd.com/show_bug.cgi?id=9956

**Review by Obihoernchen — CHANGES_REQUESTED** [review]
Very nice tutorial!
I added a few comments.

Maybe rename `slurm/slurm.md` to `slurm/README.md`  to see it in github right away?

File: slurm/slurm.md (hunk 1 above)
Comment (zbenta) [reply slurm/slurm.md]: Done.

File: slurm/slurm.md (hunk 2 above)
Comment (zbenta) [reply slurm/slurm.md]: Done

File: slurm/slurm.md (hunk 3 above)
Comment (zbenta) [reply slurm/slurm.md]: Sure, done.

File: slurm/slurm.md (hunk 4 above)
Comment (zbenta) [reply slurm/slurm.md]: You are completely right.

File: slurm/slurm.md (hunk 5 above)
Comment (zbenta) [reply slurm/slurm.md]: noob alert

File: slurm/slurm.md (hunk 13 above)
Comment (zbenta) [reply slurm/slurm.md]: noob alert

File: slurm/slurm.md (hunk 14 above)
Comment (zbenta) [reply slurm/slurm.md]: noob alert

File: slurm/slurm.md (hunk 15 above)
Comment (zbenta) [reply slurm/slurm.md]: Done

File: slurm/slurm.md (hunk 16 above)
Comment (zbenta) [reply slurm/slurm.md]: yep, it was meant to be -y

File: slurm/slurm.md (hunk 17 above)
Comment (zbenta) [reply slurm/slurm.md]: We removed it altogether, since it was just a test.

**Review by Obihoernchen — APPROVED** [review]

**Pull request merged**